iree/task/task.c

```diff
@@ -533,19 +533,20 @@ void iree_task_dispatch_issue_sharded(
   // By the task being ready to execute we know any dependencies on the
   // indirection buffer have been satisfied and its safe to read.
   if (dispatch_task->header.flags & IREE_TASK_FLAG_DISPATCH_INDIRECT) {
-    memcpy(shared_state->workgroup_count, dispatch_task->workgroup_count.ptr,
-           sizeof(shared_state->workgroup_count));
-  } else {
-    memcpy(shared_state->workgroup_count, dispatch_task->workgroup_count.value,
-           sizeof(shared_state->workgroup_count));
+    uint32_t workgroup_count_temp[3];
+    memcpy(workgroup_count_temp, dispatch_task->workgroup_count.ptr,
+           sizeof(workgroup_count_temp));
+    memcpy(dispatch_task->workgroup_count.value, workgroup_count_temp,
+           sizeof(workgroup_count_temp));
```

Comment:
I might not be awake yet...

Is the temp variable and extra `memcpy` actually needed here? Seems like you could copy from `dispatch_task->workgroup_count.ptr` to `dispatch_task->workgroup_count.value` without the intermediate stack variable?

Reply:
I might not have been awake when writing it :)
it's not required - good catch!

Reply:
(for clarity as they are unioned I'm still going to fetch the pointer to a local, and I'll ad a comment about it too)

Reply:
👍